ruby/sdk-phase-2.md

```diff
@@ -0,0 +1,605 @@
+    * .NET - Multi wait (`Workflow.WhenAny`) does not cancel immediately, and neither does `await task`
+  * For Ruby we have chosen to match all non-Python and not interrupt on cancellation on any of the calls, but we do
+    offer a `Workflow.cancellation_future` that can be used in `any_of`.
+    * This has the added benefit that `async` library will continue to work well with our futures and vice-versa.
+* We have chosen not to have additional combinators (e.g. `then` or `compose`) because these are easy to compose on
+  their own and we can add later if we want, but we want a simple API at first.
+* Nothing in the "Workflow API" returns these futures because it is easy enough to wrap.
+* Unlike some other Ruby libraries (e.g `concurrent-ruby`), we do not wrap exceptions. That means, like our other SDKs,
+  if you need to know in a `all_of` scenario which future failed, you either check the futures or put more detail into
+  the thing raised from the future block.
+  * 💭 Why not wrap exceptions? Because certain exception types are treated differently by the outer workflow failure
+    handling logic to differentiate task failure from workflow failure, and we don't want to have special
+    unwrap-from-future-fail in there. It can get confusing for users to catch too.
+* We chose not to make whether `any_of` raises be configurable. There is a lot of confusion about what return value
+  and exception situation should be across the industry in different languages. We're taking a very simple approach
+  and if users struggle with `any_of` and needing to know which future succeeded/failed, they can be more specific in
+  their return/raise inside their future block.
+  * We chose to add an `try_any_of` that does not raise on `wait`.
+    * 💭 Why not just an arg? Because it would change return value which is a confusing thing to do just based on arg.
+    * 💭 Why not `any_of!`? Because it's not necessarily more dangerous needing `!`, just a different behavior.
+* We chose not to make whether `all_of` raises be configurable. There is a lot of confusion about what return value
+  and exception situation should be across the industry in different languages. We're taking a very simple approach
+  and if users struggle with `all_of` need to wait for other futures to succeed when one fails, they need to make sure
+  none fail (can return exception instead or something).
+  * We chose to add an `try_all_of` that does not raise on `wait`.
+    * 💭 Why not just an arg? Because we have `try_any_of` and this is consistent with that.
+    * 💭 Why not `all_of!`? Because it's not necessarily more dangerous needing `!`, just a different behavior.
+
+Example of running many activities and waiting on all to complete:
+
+```ruby
+class ManyActivityWorkflow < Temporalio::Workflow
+  def execute(count)
+    futures = count.times.map do
+      Future.new do
```

Comment:
I found Temporal ruby's [implicit Futures](https://github.com/coinbase/temporal-ruby/blob/f41efb7bb0c9e755a07382b94e00a4f31bcd33c1/README.md?plain=1#L269-L270) weird, so this could work well.
However, please try to make sure that return types propagate through the future for static types.  Would be a shame to lose type safety for async.

Reply:
Static typing is still under research, but our early research has shown it would not affect design (nor would we want it to for the majority of Ruby users that don't care about it and don't deserve to be affected by it), so we are not blocking this effort.